draft-ietf-masque-h3-datagram.md

```diff
@@ -91,85 +80,13 @@ when, and only when, they appear in all capitals, as shown here.
-require any context-specific software to enable contexts. When contexts are
-supported by the implementation, their use is optional and can be selected on
-each stream. Endpoints inform their peer of whether they wish to use contexts
-via the Sec-Use-Datagram-Contexts HTTP header, see {{context-hdr}}.
-
-When contexts are used, they are identified within the scope of a given request
-by a numeric value, referred to as the context ID. A context ID is a 62-bit
-integer (0 to 2<sup>62</sup>-1).
-
-
-## Datagram Formats {#datagram-formats}
-
-When an endpoint registers a datagram context (or the lack of contexts), it
-communicates the format (i.e., the semantics and encoding) of datagrams sent
-using this context. This is acccomplished by sending a Datagram Format Type as
-part of the datagram registration capsule, see {{register-capsule}}. This type
-identifier is registered with IANA (see {{iana-format-types}}) and allows
-applications that use HTTP Datagrams to indicate what the content of datagrams
-are. Registration capsules carry a Datagram Format Additional Data field which
-allows sending some additional information that would impact the format of
-datagrams.
-
-For example, a protocol which proxies IP packets can define a Datagram Format
-Type which represents an IP packet. The corresponding Datagram Format
-Additional Data field would be empty. An extension to such a protocol that
-wishes to compress IP addresses could define a distinct Datagram Format Type
-and exchange two IP addresses via the Datagram Format Additional Data field.
-Then any datagrams with that type would contain the IP packet with addresses
-elided.
-
-
-## Context ID Allocation {#context-id-alloc}
-
-Implementations of HTTP Datagrams that support datagram contexts MUST provide
-a context ID allocation service. That service will allow applications
-co-located with HTTP to request a unique context ID that they can subsequently
-use for their own purposes. The HTTP implementation will then parse the context
-ID of incoming HTTP Datagrams and use it to deliver the frame to the
-appropriate application context.
-
-Even-numbered context IDs are client-initiated, while odd-numbered context IDs
-are server-initiated. This means that an HTTP client implementation of the
-context ID allocation service MUST only provide even-numbered IDs, while a
-server implementation MUST only provide odd-numbered IDs. Note that, once
-allocated, any context ID can be used by both client and server - only
-allocation carries separate namespaces to avoid requiring synchronization.
-Additionally, note that the context ID namespace is tied to a given HTTP
-request: it is possible for the same numeral context ID to be used
-simultaneously in distinct requests.
+coexist on a given QUIC connection. To allow this, the QUIC DATAGRAM frame
+payload starts with an encoded stream identifier that associates the datagram
+with a given QUIC stream.
```

Comment:
```suggestion
payload starts with an encoded stream identifier that associates the datagram
with a request stream.
```

You already say "encoded stream identifier" and the goal is to associate with request streams.  In WebTransport, we might use those bidirectional streams without making them requests, which will require (or at least allow) some additional checking to see that the identified stream contains a request, but that's something WebTransport can handle.